src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,87 @@
 package mate.academy.lib;
 
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+
 public class Injector {
     private static final Injector injector = new Injector();
 
     public static Injector getInjector() {
         return injector;
     }
 
+    private Map<Class<?>, Object> instances = new HashMap<>();
+
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+
+        Object clazzImplementationInstance = null;
+        Class<?> clazz = findImplementation(interfaceClazz);
+
+        if (!clazz.isAnnotationPresent(Component.class)) {
+            throw new RuntimeException("Injection failed, missing @Component annotaion on the class " + clazz.getName());
+        }
+
+        Field[] declaredFields = interfaceClazz.getDeclaredFields();
+
+        for (Field field : declaredFields) {
+            if (field.isAnnotationPresent(Inject.class)) {
+
+                Object fieldInstance = getInstance(field.getType());
+                clazzImplementationInstance = createNewInstance(clazz);
+
+                field.setAccessible(true);
+
+                try {
+                    field.set(clazzImplementationInstance, fieldInstance);
+                } catch (IllegalAccessException e) {
+                    throw new RuntimeException("Can't initialize field value. " +
+                            "Class: " + clazz.getName() + ". Field: " + field.getName());
+                }
+            }
+        }
```

Comment:
Optional: would be great to use functional style, like ```interfaceClazz.getDeclaredFields().foreach(...)```

Reply:
It's better to leave this solution